db/hybrid_logical_vector.go

```diff
@@ -383,6 +404,14 @@ func (hlv *HybridLogicalVector) setPreviousVersion(source string, version uint64
 	hlv.PreviousVersions[source] = version
 }
 
+// setMergeVersion will take a source/version pair and add it to the HLV merge versions map
+func (hlv *HybridLogicalVector) setMergeVersion(source string, version uint64) {
+	if hlv.MergeVersions == nil {
```

Comment:
I am not entirely sure why we don't require initialization of the maps and using constructor `NewHybridLogicalVector`. I wasn't sure if it was for perf reasons so I left it.

Is there a reason to have `setMergeVersion` in a loop rather than just assign the incoming merge versions to the existing hlv?

Reply:
I agree it seems inconsistent, but don't want to open that can of worms at this point.  Leaving setMergeVersion as a single entry API to match setPreviousVersion and for potential future usage.